### ISSUE TITLE
[AddSelect]: Keyboard navigation doesn't work as expected.

### ISSUE THREAD

**Issue by sangeethababu9223** [carbon-design-system/ibm-products#6024]
### Package

Carbon for IBM Products

### Description

In `SingleAddSelect ` and `MultiAddSelect `there are the following issues
1. Keyboard navigation using `Arrow Left` doesn't work as expected.
2. In second and third levels the first elements are not automatically focused.
3. Up and Down arrow navigation has issues in MultiSelect

Related to issue [#5451](https://github.com/carbon-design-system/ibm-products/issues/5451)

### Component(s) impacted

SingleAddSelect
MultiAddSelect

### Browser

Chrome

### @carbon/ibm-products (previously @carbon/ibm-cloud-cognitive) version

v2.49.0-rc.0

### Suggested Severity

Severity 3 = The problem is visible or noticeable to users but does not impede the usability or functionality. Affects minor functionality, has a workaround.

### Product/offering

NA

### CodeSandbox or Stackblitz example

NA

### Steps to reproduce the issue (if applicable)

https://ibm-products.carbondesignsystem.com/?path=/story/ibm-products-patterns-add-and-select-singleaddselect--with-hierarchy&globals=viewport:basic

https://ibm-products.carbondesignsystem.com/?path=/story/ibm-products-patterns-add-and-select-multiaddselect--with-hierarchy&globals=viewport:basic

Trying to navigate using keyboard in both the stories will show the issues. 

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/ibm-products/blob/main/.github/CODE_OF_CONDUCT.md)
- [ ] I checked the [current issues](https://github.com/carbon-design-system/ibm-products/issues) for duplicate issues
